asserts/database.go

```diff
@@ -27,22 +27,34 @@ import (
 	"fmt"
 	"os"
 	"path/filepath"
+	"strings"
+	"time"
 
 	"golang.org/x/crypto/openpgp/packet"
 
 	"github.com/ubuntu-core/snappy/helpers"
 )
 
+// PublicKey is a public key as used by the assertion database.
+type PublicKey interface {
+	PublicKeyPrim
```

Comment:
@niemeyer we could call one PublicKey and the other PublicKeyWithExpiration or TimeLimitedPublicKey, but I would prefer the more visible/used concept to have the nicer PublicKey name